### PR TITLE
Init routing

### PR DESCRIPTION
**Pull request by David-OConnor**
https://github.com/David-OConnor/seed/issues/224

Tests currently broken.

### CODE REVIEW

**Comment by MartinKavik** [comment]
WIP, but look good! 
Could you notify me before merge? I want to try some type magic - e.g. replace `-> Init<Model>` with `-> impl Init`. I don't know if it's doable or better but I want to try it and eventually compare both versions.

**Comment by David-OConnor** [comment]
CI issue was due to fmt. Have at the `impl` - that'd be more consistent with `View` and `Update`.

**Comment by MartinKavik** [comment]
How would look like standalone `init`?
i.e. instead of
```rust
fn init(_: Url, orders: &mut impl Orders<Msg>) -> Init<Model> {
    orders.perform_cmd(fetch_repository_info());
    Init::new(Model::default())
}
```
would be..?

**Comment by David-OConnor** [comment]
I was thinking it could be a closure as a closure, but on reflection, that may not be robust enough. eg

```rust
let init = Init::from_fn(|_, | {
    orders.perform_cmd(fetch_repository_info());
    Init::new(Model::default())
})
```

**Comment by MartinKavik** [comment]
And.. what about:
i.e. standalone init (without changes)
```rust
fn init(_: Url, _: &mut impl Orders<Msg>) -> Init<Model> {
    Init::new(Model::default())
}
```
One-liner:
```rust
seed::App::build(Init::default_fn(Model::default()), update, view)
```

File: src/vdom.rs [hunk 1]
```diff
@@ -53,15 +53,47 @@ pub enum ShouldRender {
...
 type SinkFn<Ms, Mdl, ElC, GMs> = fn(GMs, &mut Mdl, &mut OrdersContainer<Ms, Mdl, ElC, GMs>);
 type ViewFn<Mdl, ElC> = fn(&Mdl) -> ElC;
 type RoutesFn<Ms> = fn(routing::Url) -> Option<Ms>;
 type WindowEvents<Ms, Mdl> = fn(&Mdl) -> Vec<events::Listener<Ms>>;
 type MsgListeners<Ms> = Vec<Box<dyn Fn(&Ms)>>;
 
+/// Used for handling initial routing.
+pub enum UrlHandling {
+    PassToRoutes,
+    Nothing,
```
Comment (MartinKavik) [inline src/vdom.rs]: Why not `None`?

Comment (David-OConnor) [reply src/vdom.rs]: Agree

**Comment by David-OConnor** [comment]
Want to try `impl Init`? Ready to merge as-is, pending fixing CI.

**Comment by MartinKavik** [comment]
I'm not able to come up with better API. I see some comments in PR which we can delete, but we can remove them with some future refactors. Merge it.

**Pull request merged**